client/src/components/formHelpers/form-validators.tsx

```diff
@@ -8,8 +8,8 @@ const editorRegex =
   /repl\.?it(\.com)?\/(@|join\/)|glitch\.com\/edit\/#!|codesandbox\.io\/s\/|github\.com/;
 const fCCRegex =
   /codepen\.io\/freecodecamp|freecodecamp\.rocks|github\.com\/freecodecamp|\.freecodecamp\.org/i;
-const localhostRegex = /localhost:/;
-const httpRegex = /http(?!s|([^s]+?localhost))/;
+const localhostRegex = /localhost:|[::1]:/;
+const httpRegex = /http(?!s|([^s]+?(localhost|[::1])))/;
```

Comment:
This does not work for `http://127.0.0.1:3000`. We should use a library for this instead of coming up with the regex ourselves - there are too many valid formats.

Reply:
The only change that needs to be made is instead of a custom regex for `localhostRegex`, we should use a library to get match against `ipv4` and `ipv6`.

The `http` logic needs to remain. In your image, you misspelt `localhost`. So, the address is treated as a public address. We cannot accept `http` addresses (unless they are for the tests, and are local), because they will be displayed on certifications/profiles, and redirect to insecure websites.